app/workers/work_packages/progress/job.rb

```diff
@@ -39,4 +39,8 @@ class WorkPackages::Progress::Job < ApplicationJob
     perform_limit: 1,
     key: -> { "WorkPackagesProgressJob" }
   )
+
+  retry_on GoodJob::ActiveJobExtensions::Concurrency::ConcurrencyExceededError,
+           wait: 5.minutes,
+           attempts: :unlimited
```

Comment:
I would change this to apply to every almost every error. So it would have to be something like

`retry_on StandardError, wait: 5.minutes, attempts: :unlimited`

This has less to do with Concurrency concerns and more with the necessity of this job to be executed. On the other hand, at some point it just doesn't make sense to try again. But we don't have a pattern yet on how to handle this (e.g. by sending out a notification to an admin). So in the meantime, I would probably do 

`retry_on StandardError, wait: 5.minutes, attempts: 3`

Reply:
Long train of thought here (correct any misunderstandings please):

1. We have a `perform_limit` of 1, meaning 1 running and infinite queued.
2. We have a fixed concurrency `key`.
3. This is the base job of 3 others, that will share the same key meaning the others will wait until the current processing job is done.
4. The process of waiting will raise a concurrency error.
5. We need these jobs to run, no matter why, no matter how long it takes for it finally 
run.

Due to the shared key, we might run on the concurrency error frequently (I have no clue on how many times per <SEGMENT OF TIME> this jobs are run), so a retry on that should take care of the exponential backoff that  `GoodJob` adds by default.

Retrying on `StandardError` seems... off. It is too generic and might keep enqueuing a job that has bad arguments or, let's say the code is broken due to the mystical influence of Production DBs.

Reply:
Ok, I can follow your reasoning with the `ConcurrencyExceededError`, @mereghost, so lets add the statement just you like you proposed initially.

As for the other errors, I currently just don't know which ones to expect. But without any other specification AFAIK, the jobs will just be retried 5 times so that might work for the time being anyways.